### ISSUE TITLE
elements of computed A matrix have complex components in values

### ISSUE THREAD

**Issue by johnwlambert** [PouriaZ/GMML#1]
I'm observing that a certain portion of the time, the returned "A" matrix is not a real matrix, but rather complex, e.g. some small percentage of the time, I see an A that resembles:

A =

   1.0e+03 *

  Columns 1 through 4

   0.0010 + 0.0000i   0.0000 + 0.0000i   0.0000 + 0.0000i   0.0000 + 0.0000i
   0.0000 + 0.0000i   0.0198 + 0.0000i   0.0484 + 0.0000i   0.0429 + 0.0000i
   0.0000 + 0.0000i   0.0484 + 0.0000i   0.1331 + 0.0000i   0.1138 + 0.0000i
   0.0000 + 0.0000i   0.0429 + 0.0000i   0.1138 + 0.0000i   0.2122 + 0.0000i

Is this expected from the geodesic mean?

**Comment by PouriaZ** [comment]
No, we do not expect the resulting mean of the algorithm to be a complex matrix. This is due to a numerical error occurred in the sharp.m. I believe it does not change the performance of the algorithm; however, I should use the real part of the output matrix of sharp.m.

Now, the code MetricLearning.m is changed a little bit to handle this type of numerical error.

Thank you
